stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -332,24 +370,8 @@ end
 @inline mul!(C::Diagonal, Da::Diagonal, Db::Diagonal, alpha::Number, beta::Number) =
     _muldiag!(C, Da, Db, alpha, beta)
 
-function mul!(C::AbstractMatrix, Da::Diagonal, Db::Diagonal, alpha::Number, beta::Number)
-    _muldiag_size_check(C, Da, Db)
-    require_one_based_indexing(C)
-    mA = size(Da, 1)
-    da = Da.diag
-    db = Db.diag
-    _rmul_or_fill!(C, beta)
-    if iszero(beta)
-        @inbounds @simd for i in 1:mA
-            C[i,i] = Ref(da[i] * db[i]) .*ₛ alpha
-        end
-    else
-        @inbounds @simd for i in 1:mA
-            C[i,i] += Ref(da[i] * db[i]) .*ₛ alpha
-        end
-    end
-    return C
-end
+mul!(C::AbstractMatrix, Da::Diagonal, Db::Diagonal, alpha::Number, beta::Number) =
+    _muldiag!(C, Da, Db, alpha, beta)    
```

Comment:
```suggestion
    _muldiag!(C, Da, Db, alpha, beta)
```

Reply:
Thanks. I didn't receive any useful feedback, just some error number. But still, okay, I should have just double checked regarding trailing whitespace.

Reply:
Wait, I'm still getting
```
% sudo make test-LinearAlgebra
env: julia: No such file or directory
make: *** [check-whitespace] Error 127
```

What is that saying?

Reply:
Hmmm, that's a good idea.  I'll open a PR.

Reply:
Also, you should be careful running things like `sudo make`.  You'll end up putting files in your build tree that are owned by root, and then a normal `make` won't work in the future, because it won't be able to delete those temporary files, etc....

Reply:
Has something changed? I used my usual "workflow" which has always worked so far.

Reply:
On CI, though, the `whitespace` job doesn't build Julia first.

So, maybe check if `CI` is in the environment and set to `true`, and fall back to using the system Julia in that case.

Reply:
> Also, you should be careful running things like `sudo make`. You'll end up putting files in your build tree that are owned by root, and then a normal `make` won't work in the future, because it won't be able to delete those temporary files, etc....

Thanks for the warning. I think I once started doing that because it didn't work otherwise for a reason that I already forgot. I'll try without `sudo` and see what happens.

Reply:
https://github.com/JuliaLang/julia/pull/44658